src/lib/es2018.asynciterable.d.ts

```diff
@@ -11,9 +11,9 @@ interface SymbolConstructor {
 
 interface AsyncIterator<T, TReturn = any, TNext = undefined> {
     // NOTE: 'next' is defined using a tuple to ensure we report the correct assignability errors in all places.
-    next(...args: [] | [TNext]): Promise<IteratorResult<T, TReturn>>;
-    return?(value?: TReturn | PromiseLike<TReturn>): Promise<IteratorResult<T, TReturn>>;
-    throw?(e?: any): Promise<IteratorResult<T, TReturn>>;
+    next(...args: [] | [TNext]): Promise<IteratorResult<T, TReturn extends undefined ? TReturn : TReturn extends PromiseLike<infer UReturn> ? UReturn : TReturn>>;

```

Comment:
As I mentioned earlier, conditional types are much more restricted in terms of assignability, so I would be concerned this could break some users.

Reply:
:+1: Removed.